internal/graph/dispatch_throttling_check_resolver_test.go

```diff
@@ -54,14 +96,113 @@ func TestDispatchThrottlingCheckResolver(t *testing.T) {
+			Frequency:        1 * time.Hour,
+			DefaultThreshold: 200,
+			MaxThreshold:     210,
+		}
+		dut := NewDispatchThrottlingCheckResolver(dispatchThrottlingCheckResolverConfig)
+		defer dut.Close()
+
+		initialMockResolver := NewMockCheckResolver(ctrl)
+		dut.SetDelegate(initialMockResolver)
+
+		// this is to simulate how many times request has been dispatched
+		// Since this is a small value, we will not expect throttling
+		req := &ResolveCheckRequest{RequestMetadata: NewCheckRequestMetadata(10)}
+		req.GetRequestMetadata().DispatchCounter.Store(203)
+		response := &ResolveCheckResponse{Allowed: true, ResolutionMetadata: &ResolveCheckResponseMetadata{
+			DatastoreQueryCount: 202,
+		}}
+
+		// Here, we count how many times the resolve check is dispatched by the DUT
+		resolveCheckDispatchedCounter := 0
+		initialMockResolver.EXPECT().ResolveCheck(gomock.Any(), req).DoAndReturn(func(ctx context.Context, req *ResolveCheckRequest) (*ResolveCheckResponse, error) {
+			resolveCheckDispatchedCounter++
+			return response, nil
+		}).Times(1)
+
+		ctx := context.Background()
+		ctx = telemetry.ContextWithDispatchThrottlingThreshold(ctx, uint32(205))
+
+		_, err := dut.ResolveCheck(ctx, req)
+		require.NoError(t, err)
+		require.Equal(t, 1, resolveCheckDispatchedCounter)
+		require.False(t, req.GetRequestMetadata().WasThrottled.Load())
+	})
+
 	t.Run("above_threshold_should_be_throttled", func(t *testing.T) {
 		ctrl := gomock.NewController(t)
 		defer ctrl.Finish()
 
 		dispatchThrottlingCheckResolverConfig := DispatchThrottlingCheckResolverConfig{
 			// We set timer ticker to 1 hour to avoid it interfering with test
-			Frequency: 1 * time.Hour,
-			Threshold: 200,
+			Frequency:        1 * time.Hour,
+			DefaultThreshold: 200,
+			MaxThreshold:     200,
+		}
+		dut := NewDispatchThrottlingCheckResolver(dispatchThrottlingCheckResolverConfig)
+		defer dut.Close()
+
+		initialMockResolver := NewMockCheckResolver(ctrl)
+		dut.SetDelegate(initialMockResolver)
+
+		req := &ResolveCheckRequest{RequestMetadata: NewCheckRequestMetadata(10)}
+		req.GetRequestMetadata().DispatchCounter.Store(201)
+		response := &ResolveCheckResponse{Allowed: true, ResolutionMetadata: &ResolveCheckResponseMetadata{
+			DatastoreQueryCount: 10,
+		}}
+
+		// Here, we count how many times the resolve check is dispatched by the DUT
+		resolveCheckDispatchedCounter := 0
+
+		var goFuncDone sync.WaitGroup
+		goFuncDone.Add(1)
+
+		initialMockResolver.EXPECT().ResolveCheck(gomock.Any(), req).DoAndReturn(func(ctx context.Context, req *ResolveCheckRequest) (*ResolveCheckResponse, error) {
+			resolveCheckDispatchedCounter++
+			return response, nil
+		}).Times(1)
+
+		var goFuncInitiated sync.WaitGroup
+		goFuncInitiated.Add(1)
+
+		ctx := context.Background()
+
+		go func() {
+			defer goFuncDone.Done()
+			goFuncInitiated.Done()
+			_, err := dut.ResolveCheck(ctx, req)
+			//nolint:testifylint
+			require.NoError(t, err)
+		}()
```

Comment:
Declare `err` externally and assign it inside, then `require.NoError(t, err)` outside. For example,

```go
var resolveCheckErr error
go func() {
  _ , resolveCheckErr := dut.ResolveCheck(ctx, req)
}()
require.NoError(t, resolveCheckErr)
```

Reply:
updated